java/src/main/java/org/rocksdb/ColumnFamilyDescriptor.java

```diff
@@ -70,17 +83,23 @@ public boolean equals(final Object o) {
     }
 
     final ColumnFamilyDescriptor that = (ColumnFamilyDescriptor) o;
-    return Arrays.equals(columnFamilyName_, that.columnFamilyName_)
-            && columnFamilyOptions_.nativeHandle_ == that.columnFamilyOptions_.nativeHandle_;
+    return nativeHandle_ == that.getNativeHandle();
   }
 
   @Override
   public int hashCode() {
-    int result = (int) (columnFamilyOptions_.nativeHandle_ ^ (columnFamilyOptions_.nativeHandle_ >>> 32));
-    result = 31 * result + Arrays.hashCode(columnFamilyName_);
-    return result;
+    return 31 * ((int)(nativeHandle_ ^ (nativeHandle_ >>> 32)));
+
+  }
+  @Override
+  protected void disposeInternal(final long handle) {
+    disposeJni(nativeHandle_);
```

Comment:
I think the order of these functions calls should be swapped, and the `disposeJni` should go in a `finally` block.

Reply:
How about instead, so that we don't for this class have a native method called `disposeJni` instead of the usual name: `disposeInternal`:

```java
@Override
protected void disposeInternal() {
  try {
    if (implicitlyCreatedColumnFamilyOptions) {
      columnFamilyOptions.close();
    }
  } finally {
    super.disposeInternal();
  }
}
```

Reply:
`disposeJni` is `static`. And I can't override method with static method. This is why I have `disposeJni`

I will swap the order and add finally block.

Reply:
What I mean is that you don't need to add a disposeJni method. Other classes that extend RocksObject add a disposeInternal(nativeHandle) method - we should follow the same pattern here.